src/Aspire.Hosting.Azure.Sql/AzureSqlServerResource.cs

```diff
@@ -107,12 +114,94 @@ public override ProvisionableResource AddAsExistingResource(AzureResourceInfrast
     /// <inheritdoc/>
     public override void AddRoleAssignments(IAddRoleAssignmentsContext roleAssignmentContext)
     {
+        if (this.IsExisting())
+        {
+            // This resource is already an existing resource, so don't add role assignments
+            return;
+        }
+
         var infra = roleAssignmentContext.Infrastructure;
-        var postgres = (SqlServer)AddAsExistingResource(infra);
+        var sqlserver = (SqlServer)AddAsExistingResource(infra);
+        var isRunMode = roleAssignmentContext.ExecutionContext.IsRunMode;
 
-        var principalId = roleAssignmentContext.PrincipalId;
-        var principalName = roleAssignmentContext.PrincipalName;
+        var sqlServerAdmin = UserAssignedIdentity.FromExisting("sqlServerAdmin");
+        sqlServerAdmin.Name = AdminName.AsProvisioningParameter(infra);
+        infra.Add(sqlServerAdmin);
 
-        AzureSqlExtensions.AddActiveDirectoryAdministrator(infra, postgres, principalId, principalName);
+        // When not in Run Mode (F5) we reference the managed identity
+        // that will need to access the database so we can add db role for it
+        // using its ClientId. In the other case we use the PrincipalId.
+
+        var userId = roleAssignmentContext.PrincipalId;
+
+        if (!isRunMode)
+        {
+            var managedIdentity = UserAssignedIdentity.FromExisting("mi");
+            managedIdentity.Name = roleAssignmentContext.PrincipalName;
+            infra.Add(managedIdentity);
+
+            userId = managedIdentity.ClientId;
+        }
+
+        foreach (var database in Databases.Keys)
+        {
+            var uniqueScriptIdentifier = Infrastructure.NormalizeBicepIdentifier($"{this.GetBicepIdentifier()}_{database}");
+            var scriptResource = new SqlServerScriptProvisioningResource($"script_{uniqueScriptIdentifier}")
+            {
+                Name = BicepFunction.Take(BicepFunction.Interpolate($"script-{BicepFunction.GetUniqueString(this.GetBicepIdentifier(), roleAssignmentContext.PrincipalName, new StringLiteralExpression(database), BicepFunction.GetResourceGroup().Id)}"), 24),
+                Kind = "AzurePowerShell",
+                AZPowerShellVersion = "7.4"
```

Comment:
How do we keep this up-to-date?

Reply:
I updated it to the latest right now and added a comment with the official list. I assume we'll be able to remove it once the resource is supported in the CDK.